### ISSUE TITLE
It's not working in my case

### ISSUE THREAD

**Issue by EniddeallA** [anhnh2002/XTTSv2-Finetuning-for-New-Languages#28]
I’m encountering challenges with fine-tuning XTTS for Moroccan Darija despite having a high-quality extensive dataset.
Dataset: 1,000 hours of Moroccan Darija audio, segmented into 1–5 second clips, sampled at 16 kHz. The dataset is accurately transcribed in Arabic alphabets and aligned with audio.
Setup:  L4 GPU with 24GB VRAM

I used your script for tokenization to extend the Arabic language.
For the gpt trainer I first tried the default configuration, then started changing depending on results, I tried Multiple Schedulers for LR, and experimented with different hyperparameters(), but all lead to the Increasing of avg_loss_mel_ce after a couple epochs.

This is the [log](https://github.com/user-attachments/files/17710615/trainer_0_log.3.txt) file for the last run interupted at 50 epochs with the following config:

```
num_epochs 100 
batch_size 10 
grad_acumm 25
lr 1e-5
```
```
config.lr_scheduler = "CosineAnnealingWarmRestarts"  # Use PyTorch's warm restarts scheduler
config.lr_scheduler_params = {
    "T_0": 20,             # Number of epochs to complete the first cosine cycle
    "T_mult": 1,           # Factor to increase cycle length by after each restart
    "eta_min": 5e-6,       # Minimum learning rate at the end of each cycle
    "last_epoch": -1
}
```

I would really appreciate your help.

**Comment by EniddeallA** [comment]
Could it be a tokenization issue, I run exactly the command on your readme with language set to Arabic:
`
python extend_vocab_config.py --output_path=checkpoints/ --metadata_path datasets/metadata_train.csv --language ar --extended_vocab_size 2000
`
what do you think?

**Comment by anhnh2002** [comment]
No, it's not a tokenization issue. From what I see in your provided log file, each epoch contains 480 steps. This means you only have a maximum of 5s * 10 * 480 = 24,000 seconds, which is approximately over 6 hours of audio in your training set. This conflicts with the information you provided (1,000 hours of dataset). There is a mistake here.

**Comment by EniddeallA** [comment]
You're right, I just checked and it turns out I only uploaded 5 hours audio recordings. Do you think I can make it work with the current dataset or should I add more audio recordings?

**Comment by anhnh2002** [comment]
You need to add more audio samples, and 5 epochs are sufficient.

**Comment by cod3r0k** [comment]
Could you guide me in preparing nice tokenizer for Arabic as you mentioned? @EniddeallA

**Comment by EniddeallA** [comment]
Hello @cod3r0k, I'm not sure what do you mean but as I mentioned above, I used the `extend_vocab_config.py` script in this repository on my Arabic dataset.

**Comment by cod3r0k** [comment]
Great @EniddeallA , thanks. Do you mean that you don't change anything in the code and just run it? (Because I think the `cleaner` or `tokenizer` is for English, and I don't know whether I should change them or not, so I'm asking.)
